lib/middleware.ts

```diff
@@ -36,7 +36,13 @@ export const useAuth = createMiddleware(async (req, res, next) => {
   const shareToken = await parseShareToken(req);
 
   let user = null;
-  const { userId, authKey } = payload || {};
+  let userId = null;
+
+  const { authKey } = payload || {};
+
+  if (typeof payload === 'string') {
```

Comment:
Payload should never be a string. I think the error is in the `/pages/api/auth/login`, line 50. It should set { userId: user.id } and not simply user.id.

Reply:
fixed